packages/shared/__tests__/helpers/utilities.ts

```diff
@@ -1,6 +1,10 @@
+import { act } from '@testing-library/react-hooks';
 import { waitFor } from '@testing-library/react';
 import nock from 'nock';
 
 export const expectNockDone = (): void => expect(nock.isDone()).toBeTruthy();
 
 export const waitForNock = (): Promise<void> => waitFor(expectNockDone);
+
+export const waitForRerender = (ms = 100): Promise<void> =>
+  act(() => new Promise((resolve) => setTimeout(resolve, ms)));
```

Comment:
Is this reliable? How do we know the 100ms is enough?
Just thinking ahead will this cause issues down the line if the codebase is even bigger?

Reply:
I have tried 1ms earlier but it seems to have not worked. Now I tried it again. It worked. Oh, tests 😅 

Intuitively, `1ms` should work since our only goal is to put the things we need to test at the end of the event loop (after use effect).

Reply:
Since the sleep-like utility felt like a hack than the actual test to use. I tried other ways and properly use the `act` API and it worked!

Thank you for your awesome comment! 🚀